src/components/EmployeeList.jsx

```diff
@@ -0,0 +1,104 @@
+import axios from "axios";
+import "../styles/dashboard.css";
+import { toast } from "react-toastify";
+import { useEffect, useState } from "react";
+import { Input, Dropdown, Button } from "antd";
+const { Search } = Input;
+import { RegistrationForm } from "./RegistrationForm";
+import { UserOutlined, UserAddOutlined } from "@ant-design/icons";
+
+export const EmployeeList = () => {
+  const [employees, setEmployees] = useState([]);
+  const [showForm, setShowForm] = useState(false);
+  const [searchType, setSearchType] = useState("Everyone");
+
+  const getAllEmployees = () => {
+    axios
+      .get("http://localhost:8080/api/emp/getemployees")
+      .then((response) => {
+        setEmployees(response.data.message);
+      })
+      .catch(() => {
+        toast.error("Error fetching employees");
+      });
+  };
+
+  useEffect(() => {
+    getAllEmployees();
+  }, []);
```

Comment:
should API calls be placed in useEffect or useLayoutEffect ? Look into this, what approach is better

Reply:
sir according to what I have gathered, useEffect is preferred option for API calls as it is asynchronous and doesn't affect the rendering performance, useLayoutEffect can slow down the process if task is time consuming, rather than fetching and manipulating data its more suitable for layout related tasks